### PR TITLE
pkg/tempo: support multiple instances of tempo (BREAKING CHANGE)

### LINKED ISSUE #320: Support multiple Tempo instances
Related to #313; support multiple instances of a Tempo configuration.

### PR DESCRIPTION
**Pull request by rfratto**
⚠️  BREAKING CHANGE ⚠️ 

#### PR Description 
Supports multiple tempo instances. This is a breaking change in the same vein as #353. 

#### Which issue(s) this PR fixes 
Closes #320 

#### PR Checklist

- [x] CHANGELOG updated 
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by rfratto** [comment]
I've tested this and everything appears to be working properly.

File: pkg/loki/loki.go [hunk 1]
```diff
@@ -42,15 +42,16 @@ func New(reg prometheus.Registerer, c Config, l log.Logger) (*Loki, error) {
 			continue
 		}
 
-		r := prometheus.WrapRegistererWith(prometheus.Labels{"loki_name": ic.Name}, reg)
+		r := prometheus.WrapRegistererWith(prometheus.Labels{"loki_config": ic.Name}, reg)
+		il := log.With(l, "loki_config", ic.Name)
```
Comment (rfratto) [inline pkg/loki/loki.go:46]: Sneaking in a Loki change 🤫 I forgot to add a label to the logs to indicate which config it was coming from

cc @mattdurham

File: pkg/tempo/config.go [hunk 2]
```diff
@@ -50,23 +85,9 @@ type PushConfig struct {
...
-
-	type plain Config
-	return unmarshal((*plain)(c))
-}
-
-func (c *Config) otelConfig() (*configmodels.Config, error) {
+func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
 	otelMapStructure := map[string]interface{}{}
 
-	if !c.Enabled {
```
Comment (rfratto) [inline pkg/tempo/config.go:66]: This code path should never have hit anyway, cmd/agent doesn't instantiate the tempo subsystem unless it's enabled.

**Review by mattdurham — APPROVED** [review]

File: docs/getting-started.md [hunk 3]
```diff
@@ -250,18 +250,20 @@ loki:
...
-        grpc: # listens on the default jaeger grpc port: 14250
-  push_config:
-    endpoint: localhost:55680
-    insecure: true  # only add this if TLS is not required
-    batch:
-      timeout: 5s
-      send_batch_size: 100
-    queue:
-      retry_on_failure: true
+  configs:
```
Comment (joe-elliott) [inline docs/getting-started.md:253]: i don't necessarily disagree with this, but it's going to create a lot of copypasta boilerplate when you want to create multiple pipelines.

does prometheus/loki allow for completely independent config per remote write target?  should we just emulate those systems and have the same pipeline that writes to multiple backends?

**Review by joe-elliott — APPROVED** [review]
one thought.  code looks :+1:

File: docs/getting-started.md (hunk 3 above)
Comment (rfratto) [reply docs/getting-started.md:253]: I think that's a separate PR, actually. There's two features someone would want:

1. Configure distinct pipelines to send to distinct locations (this PR) 
2. Configure one pipeline to send to multiple locations (future PR)

The "distinct pipelines" is now supported in all of our subsystems with this PR, but the tracing subsystem still lacks a way to send spans to multiple endpoints, which I def want to also support without having people to use this `configs` block.

**Pull request merged**